debanjum/semantic-search
# Create a UI from which a user can update their config

# Incoming Changes
- Create a basic UI page which the user can utilize to update their config and regenerate file embeddings. If there are type errors in their config input, they will not be able to update the config as intended.
- Integrate pydantic.BaseModel into the config data models in order to streamline input validation and mapping

## Screenshot
![image](https://user-images.githubusercontent.com/65192171/144947990-0564dd21-8429-49a6-b62e-cf8042ed5ab0.png)

# Test
- Check out this branch and start the service
- Navigate to https://localhost:8000/ui to view the config page
  - Click on "update" to update the source config file
  - Click on "regenerate" to regenerate the embeddings

# Refer:
- https://pydantic-docs.helpmanual.io/usage/models/

Closes #7
